symbolic-symcache/src/new/writer.rs

```diff
@@ -308,6 +313,113 @@ impl SymCacheConverter {
+                    }
+
+                    let string_bytes = &mut self.string_bytes;
+                    let strings = &mut self.strings;
+                    let name_offset = Self::insert_string(string_bytes, strings, &function.name);
+
+                    let (fun_idx, _) = self.functions.insert_full(raw::Function {
+                        name_offset,
+                        comp_dir_offset: u32::MAX,
+                        entry_pc: address,
+                        lang: Language::CSharp as u32,
+                    });
+                    fun_idx
+                };
+            }
+
+            let managed_dir = if record.managed_file_info.dir_str().is_empty() {
+                None
+            } else {
+                Some(record.managed_file_info.dir_str())
+            };
+            let mut location = transform::SourceLocation {
+                file: transform::File {
+                    name: record.managed_file_info.name_str(),
+                    directory: managed_dir,
+                    comp_dir: None,
+                },
+                line: record.managed_line,
+            };
+            for transformer in &self.transformers.0 {
+                location = transformer.transform_source_location(location);
+            }
+
+            let string_bytes = &mut self.string_bytes;
+            let strings = &mut self.strings;
+            let path_name_offset = Self::insert_string(string_bytes, strings, &location.file.name);
+            let directory_offset = location
+                .file
+                .directory
+                .map_or(u32::MAX, |d| Self::insert_string(string_bytes, strings, &d));
+
+            let (file_idx, _) = self.files.insert_full(raw::File {
+                path_name_offset,
+                directory_offset,
+                comp_dir_offset: u32::MAX,
+            });
+
+            let source_location = raw::SourceLocation {
+                file_idx: file_idx as u32,
+                line: location.line,
+                function_idx: function_idx as u32,
+                inlined_into_idx: u32::MAX,
+            };
+
+            match self.ranges.entry(address) {
+                btree_map::Entry::Vacant(entry) => {
+                    entry.insert(source_location);
+                }
+                btree_map::Entry::Occupied(mut entry) => {
+                    // TODO: This exists in native-only mappings, but we don't know yet if it's
+                    // possible to generate these types of records in managed code.
+                    // println!(
+                    //     "Found what's probably an inlined source {}::{}:L{}",
+                    //     record.managed_file_info.path_str(),
+                    //     record.managed_symbol,
+                    //     record.managed_line,
+                    // );
+                    entry.insert(source_location);
+                }
+            }
```

Comment:
fyi this entire bit is largely stolen from `process_symbolic_function`, so it might be good to compare that side-by-side with this section of the diff.